### ISSUE TITLE
Turn off "Argument of type ... can not be assigned to parameter ..."

### ISSUE THREAD

**Issue by nguyentritai2906** [microsoft/pyright#2327]
I'm not sure whether Pyright or Python check for this. Can't seem to find a way to suppress this.

Here's my configuration

```lua
require'lspconfig'.pyright.setup{
    on_attach = require'modules.lsp'.on_attach,
    settings = {
        python = {
            analysis = {
                autoSearchPaths = true,
                diagnosticMode = "workspace",
                useLibraryCodeForTypes = false,
                typeCheckingMode = "off",
            },
        },
    },
}
```

Pyright gives:

```
Diagnostics:
1. Argument of type "int | list[Unknown] | Unknown" cannot be assigned to parameter "padding" of type "str | int | Tuple[int, int]" in function "__init__"
  Type "int | list[Unknown] | Unknown" cannot be assigned to type "str | int | Tuple[int, int]"
    Type "list[Unknown]" cannot be assigned to type "str | int | Tuple[int, int]"
      "list[Unknown]" is incompatible with "str"
      "list[Unknown]" is incompatible with "int"
      "list[Unknown]" is incompatible with "Tuple[int, int]"
```

Cause by:

```python
def autopad(k, p=None):
    # Pad to 'same'
    if p is None:
        p = k // 2 if isinstance(k, int) else [x // 2 for x in k]
    return p


class SharpenConv(nn.Module):
    # SharpenConv convolution
    def __init__(self, c1, c2, k=3, s=1, p=None, g=1, act=True):
        super(SharpenConv, self).__init__()
        sobel_kernel = np.array([[-1, -1, -1], [-1, 8, -1], [-1, -1, -1]], dtype='float32')
        kenel_weight = np.vstack([sobel_kernel]*c2*c1).reshape(c2,c1,3,3)
        self.conv = nn.Conv2d(c1, c2, k, s, autopad(k, p), groups=g, bias=False) # This line gives error above
        self.conv.weight.data = torch.from_numpy(kenel_weight)
        self.conv.weight.requires_grad = False
        self.bn = nn.BatchNorm2d(c2)
        try:
            self.act = nn.Hardswish() if act else nn.Identity()
        except:
            self.act = nn.Identity()

    def forward(self, x):
        return self.act(self.bn(self.conv(x)))

    def fuseforward(self, x):
        return self.act(self.conv(x))
```

**Comment by erictraut** [comment]
The `padding` parameter to `Conv2d` is annotated as accepting type `_size_2_t` which is a type alias for `str | int | tuple[int, int]`. You are attempting to pass a list to it, so this is a type violation.

If you change your code to the following, the error will go away:
```python
def autopad(k, p=None):
    # Pad to 'same'
    if p is None:
        p = k // 2 if isinstance(k, int) else (k[0] // 2, k[1] // 2)
    return p
```
